sklearn/metrics/_plot/confusion_matrix.py

```diff
@@ -0,0 +1,202 @@
+        import matplotlib.pyplot as plt
+
+        if ax is None:
+            fig, ax = plt.subplots()
+        else:
+            fig = ax.figure
+
+        cm = self.confusion_matrix
+        self.im_ = ax.imshow(cm, interpolation='nearest', cmap=cmap)
+        self.text_ = None
+
+        cmap_min, cmap_max = self.im_.cmap(0), self.im_.cmap(256)
+
+        if include_values:
+            self.text_ = np.empty_like(cm, dtype=object)
+            if values_format is None:
+                values_format = '.2f' if self.normalize else 'd'
+            thresh = (cm.max() - cm.min()) / 2.
+            for i, j in product(range(cm.shape[0]), range(cm.shape[1])):
+                color = cmap_max if cm[i, j] < thresh else cmap_min
+                self.text_[i, j] = ax.text(j, i,
+                                           format(cm[i, j], values_format),
+                                           ha="center", va="center",
+                                           color=color)
+
+        fig.colorbar(self.im_, ax=ax)
+        ax.set(xticks=np.arange(cm.shape[1]),
+               yticks=np.arange(cm.shape[0]),
+               xticklabels=self.target_names,
+               yticklabels=self.target_names,
+               ylabel="True label",
+               xlabel="Predicted label")
+
+        plt.setp(ax.get_xticklabels(), rotation=xticks_rotation)
+
+        self.figure_ = fig
+        self.ax_ = ax
+        return self
+
+
+def plot_confusion_matrix(estimator, X, y_true, sample_weight=None,
+                          labels=None, target_names=None,
+                          include_values=True, normalize=False,
+                          xticks_rotation='horizontal',
+                          values_format=None,
+                          cmap='viridis', ax=None):
+    """Plot Confusion Matrix.
+
+    Read more in the :ref:`User Guide <visualizations>`.
+
+    Parameters
+    ----------
+    estimator : estimator instance
+        Trained classifier.
+
+    X : {array-like, sparse matrix} of shape (n_samples, n_features)
+        Input values.
+
+    y : array-like of shape (n_samples,)
+        Target values.
+
+    sample_weight : array-like of shape (n_samples,), default=None
+        Sample weights.
+
+    labels : array-like of shape (n_classes,), default=None
+        List of labels to index the matrix. This may be used to reorder or
+        select a subset of labels. If `None` is given, those that appear at
+        least once in `y_true` or `y_pred` are used in sorted order.
+
+    target_names : array-like of shape (n_classes,), default=None
+        Target names used for plotting. By default, `labels` will be used if
+        it is defined, otherwise the unique labels of `y_true` and `y_pred`
+        will be used.
+
+    include_values : bool, default=True
+        Includes values in confusion matrix.
+
+    normalize : bool, default=False
+        Normalizes confusion matrix.
```

Comment:
The user might want to normalise over either axis, or altogether.

Reply:
Four options? I guess we can do `'row'`, `'column'`, `'all'`, `None`?

Reply:
I'm okay to not provide this flexibility, too. Another way to specify it is "all", "recall", "precision", None.

Reply:
Would it make sense to use "truth" and "predicted" instead of "recall" and "precision"?

Reply:
Updated PR to use 'truth' and 'predicted'. Almost feels like this should be in `confusion_matrix` itself.